rust/operator-binary/src/catalog/commons.rs

```diff
@@ -27,15 +29,43 @@ impl ExtendCatalogConfig for MetastoreConnection {
     async fn extend_catalog_config(
         &self,
         catalog_config: &mut CatalogConfig,
-        _catalog_name: &str,
-        _catalog_namespace: Option<String>,
-        _client: &Client,
+        catalog_name: &str,
+        catalog_namespace: Option<String>,
+        client: &Client,
     ) -> Result<(), FromTrinoCatalogError> {
-        catalog_config.add_configmap_property(
-            "hive.metastore.uri",
-            self.config_map.clone(),
-            "HIVE",
-        );
+        let hive_cm: ConfigMap = client
+            .get(
+                &self.config_map,
+                catalog_namespace
+                    .as_deref()
+                    .context(ObjectHasNoNamespaceSnafu)?,
+            )
+            .await
+            .with_context(|_| FailedToGetDiscoveryConfigMapSnafu {
+                catalog: catalog_name.to_string(),
+                cm_name: self.config_map.to_string(),
+            })?;
+
+        let data_key = "HIVE";
+        let hive_connection = hive_cm
+            .data
+            .as_ref()
+            .with_context(|| FailedToGetDiscoveryConfigMapDataSnafu {
+                catalog: catalog_name.to_string(),
+                cm_name: self.config_map.to_string(),
+            })?
+            .get(data_key)
+            .with_context(|| FailedToGetDiscoveryConfigMapDataKeySnafu {
+                catalog: catalog_name.to_string(),
+                cm_name: self.config_map.to_string(),
+                data_key: data_key.to_string(),
+            })?;
+
+        // This is tightly coupled now with the hive discovery config map data layout now
+        let transformed_hive_connection =
+            hive_connection.split('\n').collect::<Vec<&str>>().join(",");
```

Comment:
Nit: A simple `.replace('\n', ',')` or similar might do the trick as well

Reply:
Done https://github.com/stackabletech/trino-operator/pull/348/commits/f86d2336447005b7fa0718c8869d4af7355b8af9